### ISSUE TITLE
LinkedList jitclass example is broken

### ISSUE THREAD

**Issue by sklam** [numba/numba#1718]
```
Internal error:
TypeError: Invalid store of %"deferred.4329823704.value" to {i8*, {i32, {%"deferred.4329823704.data", i8}}*} in <numba.datamodel.models.OptionalModel object at 0x106713278>
File "linkedlist.py", line 53
```

**Comment by sklam** [comment]
The JIT class PR was merged at cc21828, where the example still works.  The example is first broken at commit ec2dc85 .

**Comment by sklam** [comment]
Problem confirmed to be mishandling of Optional to Optional casting.
